### PR TITLE
Fix #42 RADIUS authentication support for push, append, challenge/response modes

### LINKED ISSUE #42: [Customer Requested] Add support for Radius authentication
**Is your feature request related to a problem? Please describe.**
Currently, only `cyberark` and `ldap` are supported authentication methods for `cybr logon`.

**Describe the solution you'd like**
Support should be included for Radius authentication via `username+password`, `append`, and `challenge-response`.

### PR DESCRIPTION
**Pull request by infamousjoeg**

### CODE REVIEW

File: pkg/cybr/api/client.go [hunk 1]
```diff
@@ -28,7 +28,14 @@ func getUserHomeDir() (string, error) {
 
 // IsValid checks to make sure that the authentication method chosen is valid
 func (c *Client) IsValid() error {
-	if c.AuthType == "cyberark" || c.AuthType == "ldap" {
+	if c.AuthType == "cyberark" ||
+		c.AuthType == "ldap" ||
```
Comment (AndrewCopeland) [inline pkg/cybr/api/client.go:32]: Why do we have 3 different authtypes when they boil down to 'radius'?

File: cmd/logoff.go [hunk 2]
```diff
@@ -21,17 +21,13 @@ var logoffCmd = &cobra.Command{
...
-			log.Fatalf("Failed to log off. %s", err)
-			return
-		}
 		// Remove the config file written to local file system
 		err = client.RemoveConfig()
 		if err != nil {
 			log.Fatalf("Failed to remove configuration file. %s", err)
 		}
+		// Logoff the PAS REST API
+		_ = client.Logoff()
```
Comment (AndrewCopeland) [inline cmd/logoff.go:30]: Is this intended to not catch error?

File: pkg/cybr/api/auth.go [hunk 3]
```diff
@@ -23,9 +30,23 @@ func (c *Client) Logon(req LogonRequest) error {
...
+			byteOTPCode, err := terminal.ReadPassword(int(syscall.Stdin))
+			req.Password = string(byteOTPCode)
+			fmt.Println()
+			if err != nil {
+				log.Fatalln("An error occurred trying to read one-time passcode from " +
+					"Stdin. Exiting...")
+			}
+			token, err = httpJson.SendRequestRaw(url, "POST", "", req, c.InsecureTLS, c.Logger)
+			return nil
+		}
```
Comment (AndrewCopeland) [inline pkg/cybr/api/auth.go:49]: Line of sight is a little wonky here. I think we need some refactoring. Also prompting for interactive logon in the api package makes me a little hesitant. Let me chew on this one for a little.

File: pkg/cybr/api/auth.go [hunk 4]
```diff
@@ -2,9 +2,16 @@ package api
...
+	"log"
 	"strings"
+	"syscall"
 
 	httpJson "github.com/infamousjoeg/cybr-cli/pkg/cybr/helpers/httpjson"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+var (
+	token []byte
```
Comment (AndrewCopeland) [inline pkg/cybr/api/auth.go:14]: Why is this here now?

**Review by AndrewCopeland — CHANGES_REQUESTED** [review]
Left some comments

File: pkg/cybr/api/auth.go [hunk 5]
```diff
@@ -23,9 +30,23 @@ func (c *Client) Logon(req LogonRequest) error {
...
+			// Get secret value from STDIN
+			fmt.Print("Enter one-time passcode: ")
+			byteOTPCode, err := terminal.ReadPassword(int(syscall.Stdin))
+			req.Password = string(byteOTPCode)
+			fmt.Println()
+			if err != nil {
+				log.Fatalln("An error occurred trying to read one-time passcode from " +
+					"Stdin. Exiting...")
+			}
+			token, err = httpJson.SendRequestRaw(url, "POST", "", req, c.InsecureTLS, c.Logger)
```
Comment (AndrewCopeland) [inline pkg/cybr/api/auth.go:47]: What kind of token is returned here? Is it never set as `c.SessionToken`

File: cmd/logoff.go (hunk 2 above)
Comment (infamousjoeg) [reply cmd/logoff.go:30]: Yes, because during testing I found that if a user was _already_ logged off somehow, it would not remove the config file.  So I have to bypass the logoff in order to ensure the config always is removed. At which point, it'll be okay if an error occurred to allow the user to naturally time out.

**Comment by AndrewCopeland** [comment]
This looks good to me
